### ISSUE TITLE
Version 1.6.0 increased size considerably

### ISSUE THREAD

**Issue by unindented** [ReactTraining/react-media#83]
I think caused by the addition of `prop-types`:

<img width="1112" alt="screen shot 2018-02-18 at 9 26 56 am" src="https://user-images.githubusercontent.com/69485/36354746-e7525c7e-148d-11e8-82d3-03c87fa8d177.png">

People using Flow/TypeScript don't use PropTypes. Would you consider removing them from `dependencies`, and stripping them out during build or similar?

**Comment by mjackson** [comment]
Yes, thanks for bringing this to my attention. :)

Which build are you using?

**Comment by unindented** [comment]
I think the one at the root, cause `package.json` doesn't specify a `module` key, so I'm pretty sure Webpack is picking the `index.js` and `Media.js` files at the root.

**Comment by theKashey** [comment]
Current bundle size is not 2kb, as bundle-phobia displays. It is __30 kb__, as long all the used libraries, including prop-types, are inlined by rollup.

**Comment by edorivai** [comment]
@theKashey I think you're right! Even though `package.json` now specifies the `module` field, which points to the `esm` build, it seems that the `esm` build indeed does include all dependencies inline!

The bundle stats of our production build shows the following stats for `react-media`:
- Stat size: 36.47 KB
- Parsed size: 4.59 KB
- GZipped size: 1.94 KB

I suggest we make `prop-types` a `peerDependency`, and add it to the `external` list in the rollup config. As for the remaining dependencies (`invariant` and `json2mq`), I think these are small enough to keep them inline.

**Comment by edorivai** [comment]
Just leaving this here for reference: https://github.com/facebook/create-react-app/issues/209

**tl;dr**
Even if we specify `prop-types` as a `peerDependency`, it will still end up in the final bundle. There is a [babel plugin](https://www.npmjs.com/package/babel-plugin-transform-react-remove-prop-types) that strips `prop-types` entirely, but I think that should be configured by consumers of this library, not by the library itself.

**Comment by edorivai** [comment]
An update as of v1.9.2:

![image](https://user-images.githubusercontent.com/765358/49998093-a1862a80-ff93-11e8-831f-0c04665f0f4c.png)

## v1.5.1
- minified: 2.8 KB
- gzipped: 1.2 KB

## 1.9.2
- minified: 3.5 KB
- gzipped: 1.6 KB

## But...
Assuming you have your build process set up correctly (meaning you set `NODE_ENV=production`), and you are already using `@babel/runtime` in you app, we can leave out:
`prop-types`: ~800 B
`@babel/runtime`: ~300 B

Which should leave us with ~2.4 KB minified. I think that should close this issue.
